vumi/middleware/base.py

```diff
@@ -176,7 +177,19 @@ class MiddlewareStack(object):
     """
 
     def __init__(self, middlewares):
-        self.middlewares = middlewares
+        self.consume_middlewares = self._sort_by_priority(
+            middlewares, 'consume_priority')
+        self.publish_middlewares = self._sort_by_priority(
+            reversed(middlewares), 'publish_priority')
```

Comment:
We should still have the same behaviour if the configs don't specify priorities, otherwise we need to update all middleware configs in lockstep with the code change.